crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -168,14 +215,14 @@ impl MessageReceiver<MqttMessage> for MqttMessageBox {
 
 pub struct MqttActor {
     mqtt_config: mqtt_channel::Config,
-    messages: MqttMessageBox,
+    messages: Option<MqttMessageBox>,
```

Comment:
It's annoying that this is required, but we need ownership of the MqttMessageBox to split it into chunks, but `run` takes `&mut self`. If `run` is called multiple times, this will cause a panic, although we don't currently ever call run multiple times.

I definitely won't do it in this PR, but it seems reasonable to modify `Actor::run` to take `self` instead of `&mut self` to avoid this issue (and clarify that `run` is terminal, which I think is assumed in a bunch of implementations). From a quick attempt at doing this, it seems that this change causes issues when calling `run` on a `Box<dyn Actor>` (which I'm guessing is the reason it currently takes `&mut self`, but I've found [a mitigation for this problem](https://users.rust-lang.org/t/call-consuming-method-for-dyn-trait-object/69596/7), so I'm planning to create a PR that makes this change).